docs/cli/ilab-processes.md

```diff
@@ -0,0 +1,53 @@
+# Processes in InstructLab
+
+The ability to detach from processes is crucial to the user experience of InstructLab. However, the concept of multi-processing, process management, and the monitoring of processes is very complex.
+
+It is important to try and add this concept in as simply as possible, expanding on the state reporting, logging, and other features as we go along.
+
+## Phased approach to InstructLab Processes
+
+This document is going to describe phase 1 of implementing processes in InstructLab. Phase 1 is to be described as the "ilab simple process management system". This will depend purely on python packages, PID tracking, and log files to create the experience of detachable processes. The key here is the concept of the UUID, allowing a future REST API to keep track of InstructLab processes using these unique identifiers. 
+
+We can re-visit all this in phase 2, when we discuss if we want to utilize something lils systemd or a more in-depth process-monitor repo to track processes.
+
+### Phase 1
+
+Phase one would focus on adding the concept of detaching from processes. This would be implemented in `ilab 0.23.0`.
+
+The workflow would allow for:
+
+`ilab data generate -dt` (run a detached generation process)
+`ilab model train -dt` (run a detached training process)
+
+`ilab process list`
+
+```console=
++----------------------------------------+------------+-------------+---------+
+| Proccess UUID                          | Type       | Duration    | Status  |
++----------------------------------------+--------------------------+---------+
+| train-2423813fhthfbnaj                 | training   | 8 hr 32 min | Running |
+| generation-6768769599sdbjklm           | generation | 3 hr 15 min | Running |
+| generation-1345661hlsmma               | generation | 22 min      | Errored |
++----------------------------------------+---------------------+--------------+
+
+```
+
+`ilab process attach <UUID>`
+
+This command would re-attach to the given process, allowing to user to view the live logs of the process.
```

Comment:
Does this mean the process outputs both, to stdout as well as a log file at all times?

Reply:
we would tail the log file, meaning that its only really writing to the file but we are watching what is being written when the user attaches, and we listen for ctrl+c to kill the process

Reply:
I guess we always output to a log and "attaching" just moves the terminal to essentially `tail -f` that log? Except we still would need users to be able to `Crtl-C` and such

Reply:
Some of this seems to be covered below

Reply:
added a small section